jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/ByteAccumulator.java

```diff
@@ -22,18 +22,48 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.eclipse.jetty.io.ByteBufferPool;
 import org.eclipse.jetty.util.BufferUtil;
 import org.eclipse.jetty.websocket.api.MessageTooLargeException;
 
 public class ByteAccumulator
 {
-    private final List<byte[]> chunks = new ArrayList<>();
+    private final List<ByteBuffer> chunks = new ArrayList<>();
     private final int maxSize;
     private int length = 0;
+    private final ByteBufferPool bufferPool;
 
     public ByteAccumulator(int maxOverallBufferSize)
+    {
+        this(maxOverallBufferSize, null);
+    }
+
+    public ByteAccumulator(int maxOverallBufferSize, ByteBufferPool bufferPool)
     {
         this.maxSize = maxOverallBufferSize;
+        this.bufferPool = bufferPool;
+    }
+    
+    public void copyChunk(ByteBuffer buf)
+    {
```

Comment:
I think we need to check that the buf here was the one last returned from `newByteBuffer`